java-spiffe-core/src/main/java/spiffe/workloadapi/WorkloadApiClient.java

```diff
@@ -0,0 +1,240 @@
+        cancellableContext = Context.current().withCancellation();
+        cancellableContext.run(() -> workloadApiAsyncStub.fetchX509SVID(newX509SvidRequest(), streamObserver));
+        this.cancellableContexts.add(cancellableContext);
+    }
+
+    /**
+     * One-shot fetch call to get a SPIFFE JWT-SVID.
+     *
+     * @param subject       a SPIFFE ID
+     * @param audience      the audience of the JWT-SVID
+     * @param extraAudience the extra audience for the JWT_SVID
+     *
+     * @return an instance of a {@link JwtSvid}
+     *
+     * @throws //TODO: declare thrown exceptions
+     */
+    public JwtSvid fetchJwtSvid(SpiffeId subject, String audience, String... extraAudience) {
+        throw new NotImplementedException("Not implemented");
+    }
+
+    /**
+     * Fetches the JWT bundles for JWT-SVID validation, keyed by trust domain.
+     *
+     * @return an instance of a {@link JwtBundleSet}
+     * @throws //TODO: declare thrown exceptions
+     */
+    public JwtBundleSet fetchJwtBundles() {
+        throw new NotImplementedException("Not implemented");
+    }
+
+    /**
+     * Validates the JWT-SVID token. The parsed and validated
+     * JWT-SVID is returned.
+     *
+     * @param token    JWT token
+     * @param audience audience of the JWT
+     * @return the {@link JwtSvid} if the token and audience could be validated.
+     *
+     * @throws //TODO: declare thrown exceptions
+     */
+    public JwtSvid validateJwtSvid(String token, String audience) {
+        throw new NotImplementedException("Not implemented");
+    }
+
+    /**
+     * Watches for JWT bundles updates.
+     *
+     * @param jwtBundlesWatcher receives the update for JwtBundles.
+     */
+    public void watchJwtBundles(Watcher<JwtBundleSet> jwtBundlesWatcher) {
+        throw new NotImplementedException("Not implemented");
+    }
+
+    /**
+     * Closes this Workload API closing the underlying channel and cancelling the contexts.
+     */
+    @Override
+    public void close() {
+        log.info("Closing WorkloadAPI client");
+        synchronized (this) {
+            for (val context : cancellableContexts) {
+                context.close();
+            }
+            log.info("Shutting down Managed Channel");
+            managedChannel.shutdown();
+        }
+    }
+
+    private X509SVIDRequest newX509SvidRequest() {
+        return X509SVIDRequest.newBuilder().build();
+    }
+
+    private X509Context processX509Context() {
+        try {
+            Iterator<X509SVIDResponse> x509SVIDResponse = workloadApiBlockingStub.fetchX509SVID(newX509SvidRequest());
+            if (x509SVIDResponse.hasNext()) {
+                return GrpcConversionUtils.toX509Context(x509SVIDResponse.next());
+            }
+        } catch (Exception e) {
+            throw new X509ContextException("Error processing X509Context", e);
+        }
+        throw new X509ContextException("Error processing X509Context: x509SVIDResponse is empty");
+    }
```

Comment:
nit: would the following slightly modified version of your method make sense to you?
```suggestion
    private X509Context processX509Context() {
        try {
            Iterator<X509SVIDResponse> x509SVIDResponse = workloadApiBlockingStub.fetchX509SVID(newX509SvidRequest());
            if (!x509SVIDResponse.hasNext()) {
                throw new X509ContextException("Error processing X509Context: x509SVIDResponse is empty");
            }
            return GrpcConversionUtils.toX509Context(x509SVIDResponse.next());
        } catch (Exception e) {
            throw new X509ContextException("Error processing X509Context", e);
        }
    }
```